### PR TITLE
Problem: missing verification api for wc 2.0 (fix #474)

### LINKED ISSUE #474: Problem: missing verification api for wc 2.0
add verification

### PR DESCRIPTION
**Pull request by leejw51crypto**
feat: enable personal signing in web3_v2.rs example
feat: Add personal signature verification method
feat: Update wallet_connect.cc to use Walletconnect2Client
feat: Update wallet_connect.cc to use Walletconnect2Client

### CODE REVIEW

**Comment by leejw51crypto** [comment]
working on ci issues

File: demo/examples/src/wallet_connect.cc [hunk 1]
```diff
@@ -139,10 +111,14 @@ int main(int argc, char *argv[]) {
         if (test_personal) {
             /* message signing */
             rust::Vec<uint8_t> sig1 = client->sign_personal_blocking(
-                "hello", result.addresses[0].address);
+                "hello", result.eip155.accounts[0].address.address);
```
Comment (rarcifa) [inline demo/examples/src/wallet_connect.cc]: is the first address here an object that contains the value for address? address.address seems strange

File: demo/examples/src/wallet_connect.cc [hunk 2]
```diff
@@ -139,10 +111,14 @@ int main(int argc, char *argv[]) {
...
             std::cout << "signature=" << bytes_to_hex_string(sig1).c_str()
                       << std::endl;
             std::cout << "signature length=" << sig1.size() << std::endl;
+
+            bool verifyresult = client->verify_personal_blocking(
+                "hello", sig1, result.eip155.accounts[0].address.address);
```
Comment (rarcifa) [inline demo/examples/src/wallet_connect.cc]: here as well

**Review by rarcifa — APPROVED** [review]
All looks good. The only thing I am a bit confused is .address.address. Is this the new syntax to retrieve an address?

File: demo/examples/src/wallet_connect.cc (hunk 1 above)
Comment (leejw51crypto) [reply demo/examples/src/wallet_connect.cc]: yes, i'll add comment

File: demo/examples/src/wallet_connect.cc (hunk 2 above)
Comment (leejw51crypto) [reply demo/examples/src/wallet_connect.cc]: ok

**Pull request merged**